### ISSUE TITLE
Link to anchor doesn't work

### ISSUE THREAD

**Issue by DenisNeustroev** [nozzle/react-static#210]

**Comment by EmilTholin** [comment]
Hi @DenisNeustroev! More information is needed in order for anyone to help you. Do you have a minimal reproducible example?

**Comment by denis-sokolov** [comment]
# Workaround
For `#foo` and external links use a simple `<a href>` over a `Link`.

# Analysis 
Here’s the simplest example:

```jsx
import { Link } from "react-static";
<div>
    <Link to="#foo">Go to foo broken</Link>
    <a href="#foo">Go to foo working</a>
    <div style={{ height: "150vh" }} />
    <div id="foo">Foo</div>
</div>
```

Even absolute links do not work for me: `<Link to="http://example.com">Example</Link>` changes the url to `http://localhost:3000/http://example.com`. I run react-static 4.4.0, react-router-dom 4.2.2, react-router 4.2.0.

It seems to be [a design decision in react-router](https://github.com/pshrmn/react-router/blob/7309a507bda66c24411a4a722df4015a8e9db0c9/packages/react-router-dom/modules/Link.js#L73). After all, for anchor links and external links, a `Link` [has no functionality](https://reacttraining.com/react-router/web/api/Link) over an `<a>`. So a current workaround is to use `<a>` for anchor and external links.

In my opinion, it’s bad user experience. At least when I import `Link` from `react-router`, I can expect it to be related to the routing, so it’s for smart router navigation. I would fix it in react-router, but at least it’s understandable. But when I import `Link` from `react-static`, the relation to routing is lost entirely, and now the Link is pure broken.

As a fix, I would wrap the component in a simple component of our own:

```jsx
import { Link, LinkProps, NavLink, NavLinkProps } from 'react-router-dom';

function isRoutingUrl(to: LinkProps['to']) {
  if (typeof to !== 'string') return true;
  return !to.match(/^#/) && !to.match(/^[a-z]{1,10}:\/\//);
}

export function Link(props: LinkProps) {
  if (isRoutingUrl(props.to)) return <Link {...props} />
  return <a href={props.to}>{props.children}</a>;
}

export function NavLink(props: NavLinkProps) {
  if (isRoutingUrl(props.to)) return <NavLink {...props} />
  return <a href={props.to}>{props.children}</a>;
}
```

**Comment by tannerlinsley** [comment]
Yep. You'll have to roll your own Link component or just use a standard a href hash.

**Comment by denis-sokolov** [comment]
Wait wait wait, @tannerlinsley, why did you close this? Regardless of the technical explanation, there's a bug in react-static. Its links don't work! `import { Link } from 'react-static'` renders components that silently don't work.

Could we discuss UX approaches to fix this rather than rely on obtuse technical reasons behind it?

**Comment by tannerlinsley** [comment]
Sorry I misunderstood. So you want react static to be responsible for hash routing? The external link I can understand. I'll get that fixed.

**Comment by denis-sokolov** [comment]
I’m saying when a developer creates a website with react-static, the components `<Link to="#foo">` and `<Link to="http://example.com">` should work as expected in web development. I proposed a fix in a previous comment.

If you think this is outside of react-static’s responsibility, I would disagree, but understand. But then at least throw console warnings for those two use cases. Otherwise the components are silently failing, the worst kind of failure.

If you argue that we are merely exposing react-router’s component and we don’t have any responsibility beyond that, I strongly disagree. Once the component is exposed under the react-static namespace, there are different developer expectations compared to react-router namespace.

I am more than happy to propose a PR to save you the effort once you make your decision.

**Comment by tannerlinsley** [comment]
I agree. There's definitely an added expectation from the context of react
static vs react router. We should make these additions you proposed to both
the Link and NavLink components. 🤘

**Comment by denis-sokolov** [comment]
\o/

Since I have made the most noise about this, I will make a PR.

**Comment by denis-sokolov** [comment]
Closed in #231.
